src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -706,13 +708,42 @@ static int get_pool_offset(sd_dhcp_server *server, be32_t requested_ip) {
         return be32toh(requested_ip & ~server->netmask) - server->pool_offset;
 }
 
+static int prepare_new_lease(
+                DHCPLease **ret_lease,
+                be32_t address,
+                const DHCPClientId *client_id,
+                const uint8_t *chaddr,
```

Comment:
maybe use `const uint8_t chaddr[static ETH_ALEN]` to tell the compiler the expected size of the array passed here

Reply:
Actually it's supposed to be `void` since the function can be called for other types of client id. My mistake